src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -853,14 +853,21 @@ def __call__(
 
     # Overrride to properly handle the loading and unloading of the additional text encoder.
     def load_lora_weights(self, pretrained_model_name_or_path_or_dict: Union[str, Dict[str, torch.Tensor]], **kwargs):
-        state_dict, network_alpha = self.lora_state_dict(pretrained_model_name_or_path_or_dict, **kwargs)
-        self.load_lora_into_unet(state_dict, network_alpha=network_alpha, unet=self.unet)
+        unet_state_keys = self.unet.state_dict().keys()
+
+        state_dict, network_alphas = self.lora_state_dict(
+            pretrained_model_name_or_path_or_dict,
+            unet_config=self.unet.config,
```

Comment:
Should be optional for diffusers format no? IIUC the `unet_config` is only used for SGM -> Diffusers mapping part in `load_lora_weights()`.

Reply:
yeah exactly we should probably auto detect somehow if needed